### ISSUE TITLE
use preferred quote-style when generating imports

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#49559]
Import quote style should agree with enabled lints.

* `prefer_single_quotes` vs.
* `prefer_double_quotes`

I'm inclined to think the unspecified default should be single quotes (current behavior).

/fyi @lrhn

**Comment by bwilkerson** [comment]
Agreed. As far as I can remember, there have been no complaints about the default.

**Comment by lrhn** [comment]
One possible alternative is to use the quoting of existing imports, if there are any, and only fall back on single quotes if there are no imports at all.

**Comment by bwilkerson** [comment]
Yes, that's another valid signal. I think that if there's a signal in the form of an enabled lint that that should take precedence over the signal from existing code, but the signal from existing code should take precedence over the default.

**Comment by asashour** [comment]
https://dart-review.googlesource.com/c/sdk/+/260523

I am not sure where else "generating string" should be tackled.

**Comment by bwilkerson** [comment]
I'm not sure either. We can either look at every refactoring and correction producer, or wait for user feedback.

**Comment by FMorschel** [comment]
I see that both CL's have been merged. Maybe I lost something here but I don't think there is anything that this issue is tracking still. Can this be closed, @pq?

**Comment by pq** [comment]
Yes, thanks!
